routers/user/oauth.go

```diff
@@ -193,6 +212,47 @@ func newAccessTokenResponse(grant *models.OAuth2Grant, clientSecret string) (*Ac
 	}, nil
 }
 
+type userInfoResponse struct {
+	Sub      string `json:"sub"`
+	Name     string `json:"name"`
+	Username string `json:"preferred_username"`
+	Email    string `json:"email"`
+	Picture  string `json:"picture"`
+}
+
+// InfoOAuth manages request for userinfo endpoint
+func InfoOAuth(ctx *context.Context) {
+	header := ctx.Req.Header.Get("Authorization")
+	auths := strings.Fields(header)
+	if len(auths) != 2 || auths[0] != "Bearer" {
+		ctx.HandleText(http.StatusUnauthorized, "no valid auth token authorization")
+		return
+	}
+	uid := sso.CheckOAuthAccessToken(auths[1])
+	if uid == 0 {
+		handleBearerTokenError(ctx, BearerTokenError{
+			ErrorCode:        BearerTokenErrorCodeInvalidToken,
+			ErrorDescription: "Access token not assigned to any user",
+		})
+		return
+	}
+	if uid != 0 {
```

Comment:
If execution reaches this point, then it can be assumed that uid is not 0 (since if it were, then the above `return` would be hit), so you don't need to wrap the below in an if

Reply:
Removed the unneeded if statement.